### PR TITLE
[1LP][RFR][NOTEST] Added default template parameter in create_vm fixture.

### PR DESCRIPTION
**Pull request by sbulage**
## Purpose or Intent

If no template parameter provided then use DSL template.
A small enhancement in PR #9850 
- [x] Add default template parameter to `create_vm` fixture.

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 3c3e49542b95bc09102823c304341359cb8a9011

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**
    - *ansible_linked_vm_action*
    - *ansible_policy_linked_vm*
- **cfme/tests/automate/test_common_methods.py**
    - *test_vm_retire_extend*
    - *test_move_vm_into_folder*
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/automate/test_vmware_methods.py**
- **cfme/tests/cloud/test_cloud_events.py**
    - *test_vm_capture*
- **cfme/tests/cloud/test_cloud_timelines.py**
    - *mark_vm_as_appliance*
    - *control_policy*
    - *azone*
    - *test_cloud_timeline_create_event*
    - *test_cloud_timeline_policy_event*
    - *test_cloud_timeline_stop_event*
    - *test_cloud_timeline_start_event*
    - *test_cloud_timeline_diagnostic*
    - *test_cloud_timeline_rename_event*
    - *test_cloud_timeline_delete_event*
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*
- **cfme/tests/cloud_infra_common/test_discovery_and_support.py**
    - *test_vm_discovery*
- **cfme/tests/cloud_infra_common/test_genealogy.py**
- **cfme/tests/cloud_infra_common/test_html5_vm_console.py**
    - *test_html5_vm_console*
- **cfme/tests/cloud_infra_common/test_power_control_rest.py**
    - *vm_name*
    - *wait_for_vm_state_change*
    - *test_stop_vm_rest*
    - *test_start_vm_rest*
    - *test_suspend_vm_rest*
    - *test_reset_vm_rest*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/fixtures/vm.py [hunk 1]
```diff
@@ -100,20 +100,16 @@ def _get_vm_name(request):
 
 @pytest.fixture(scope="module")
 def create_vm_modscope(setup_provider_modscope, request, provider):
-    if request.param:
-        template_type = request.param
-    else:
-        pytest.error('Any appropriate Template was not passed to the fixture.')
+    # If no template parameter is passed then use DSL.
+    template_type = getattr(request, 'param', 'damnSmallLinux')
```
Comment (kedark3) [inline cfme/fixtures/vm.py]: Thanks for the PR Satya. This was a topic that we had discussed here: https://github.com/ManageIQ/integration_tests/pull/9186#discussion_r316870496 briefly. 
Do you still think we need the default? Not having a default was a decision we made to force contributors to make a concious choice of templates.

Comment (sbulage) [reply cfme/fixtures/vm.py]: I have went through the discussion and I got to know the purpose. But if you look into the changes I have made there are not conflicting with the actual implementation.
We also have many tests which only requires a basic template(DSL), so in that case we have to pass/parameterize each tests which I think is not that easy. :smile: 
What's say .?

Comment (kedark3) [reply cfme/fixtures/vm.py]: You have a good point too. I'm okay going ahead if we have reviewers consensus.

Comment (mshriver) [reply cfme/fixtures/vm.py]: I wouldn't agree with trying to pick a specific template name like this, hardcoded.

If anything, it should default to looking at the provider yaml for a small_template definition, and using that as the default when one isn't parametrized.

Comment (john-dupuy) [reply cfme/fixtures/vm.py]: Moving to WIPTEST per ^ comment.

Comment (sbulage) [reply cfme/fixtures/vm.py]: @mshriver I have added `small_template` in place of DSL.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 3b9decdd71549a9f24a0bed06824f5d1b8d5e4ba

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**
    - *ansible_linked_vm_action*
    - *ansible_policy_linked_vm*
- **cfme/tests/automate/test_common_methods.py**
    - *test_vm_retire_extend*
    - *test_move_vm_into_folder*
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/automate/test_vmware_methods.py**
- **cfme/tests/cloud/test_cloud_events.py**
    - *test_vm_capture*
- **cfme/tests/cloud/test_cloud_timelines.py**
    - *mark_vm_as_appliance*
    - *control_policy*
    - *azone*
    - *test_cloud_timeline_create_event*
    - *test_cloud_timeline_policy_event*
    - *test_cloud_timeline_stop_event*
    - *test_cloud_timeline_start_event*
    - *test_cloud_timeline_diagnostic*
    - *test_cloud_timeline_rename_event*
    - *test_cloud_timeline_delete_event*
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*
- **cfme/tests/cloud_infra_common/test_discovery_and_support.py**
    - *test_vm_discovery*
- **cfme/tests/cloud_infra_common/test_genealogy.py**
- **cfme/tests/cloud_infra_common/test_html5_vm_console.py**
    - *test_html5_vm_console*
- **cfme/tests/cloud_infra_common/test_power_control_rest.py**
    - *vm_name*
    - *wait_for_vm_state_change*
    - *test_stop_vm_rest*
    - *test_start_vm_rest*
    - *test_suspend_vm_rest*
    - *test_reset_vm_rest*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/fixtures/vm.py [hunk 2]
```diff
@@ -100,20 +100,16 @@ def _get_vm_name(request):
 
 @pytest.fixture(scope="module")
 def create_vm_modscope(setup_provider_modscope, request, provider):
-    if request.param:
-        template_type = request.param
-    else:
-        pytest.error('Any appropriate Template was not passed to the fixture.')
+    # If no template parameter is passed then use DSL.
```
Comment (john-dupuy) [inline cfme/fixtures/vm.py]: `DSL` is not always the default `small_template`, maybe just
```suggestion
    # If no template parameter is passed, use 'small_template'.
```

Comment (sbulage) [reply cfme/fixtures/vm.py]: Okay Updating it.

Comment (sbulage) [reply cfme/fixtures/vm.py]: Updated.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 0721d94a2cc8e8f28f988e434eb501ec2a374d7b

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**
    - *ansible_linked_vm_action*
    - *ansible_policy_linked_vm*
- **cfme/tests/automate/test_common_methods.py**
    - *test_vm_retire_extend*
    - *test_move_vm_into_folder*
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/automate/test_vmware_methods.py**
- **cfme/tests/cloud/test_cloud_events.py**
    - *test_vm_capture*
- **cfme/tests/cloud/test_cloud_timelines.py**
    - *mark_vm_as_appliance*
    - *control_policy*
    - *azone*
    - *test_cloud_timeline_create_event*
    - *test_cloud_timeline_policy_event*
    - *test_cloud_timeline_stop_event*
    - *test_cloud_timeline_start_event*
    - *test_cloud_timeline_diagnostic*
    - *test_cloud_timeline_rename_event*
    - *test_cloud_timeline_delete_event*
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*
- **cfme/tests/cloud_infra_common/test_discovery_and_support.py**
    - *test_vm_discovery*
- **cfme/tests/cloud_infra_common/test_genealogy.py**
- **cfme/tests/cloud_infra_common/test_html5_vm_console.py**
    - *test_html5_vm_console*
- **cfme/tests/cloud_infra_common/test_power_control_rest.py**
    - *vm_name*
    - *wait_for_vm_state_change*
    - *test_stop_vm_rest*
    - *test_start_vm_rest*
    - *test_suspend_vm_rest*
    - *test_reset_vm_rest*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**